tool/gravity/cli/clusterupdate.go

```diff
@@ -183,8 +185,86 @@ func executeUpdatePhaseForOperation(env *localenv.LocalEnvironment, environ Loca
 		return trace.Wrap(err)
 	}
 	defer updater.Close()
-	err = updater.RunPhase(context.TODO(), params.PhaseID, params.Timeout, params.Force)
-	return trace.Wrap(err)
+	return executeOrForkPhase(env, updater, params, operation)
+}
+
+// gravityResumeServiceName is the name of systemd service that executes
+// the gravity resume command.
+const gravityResumeServiceName = "gravity-resume.service"
+
+// executeOrForkPhase either directly executes the specified operation phase,
+// or launches a one-shot systemd service that executes it in the background.
+func executeOrForkPhase(env *localenv.LocalEnvironment, updater updater, params PhaseParams, operation ops.SiteOperation) error {
+	// If given the --block flag, we're running as a systemd unit (or a user
+	// requested the command to execute in foreground), so proceed to perform
+	// the command (resume or single phase) directly.
+	if params.Block {
+		return updater.RunPhase(context.TODO(),
+			params.PhaseID,
+			params.Timeout,
+			params.Force)
+	}
+	// Before launching the service, perform a few prechecks, for example to
+	// make sure that the operation is being resumed from the correct node.
+	//
+	// TODO(r0mant): Also, make sure agents are running on the cluster nodes:
+	// https://github.com/gravitational/gravity/issues/1667
+	if err := updater.Check(params.toFSM()); err != nil {
+		return trace.Wrap(err)
+	}
+	// Make sure to launch the unit command with the --block flag.
+	args := append(os.Args[1:], "--debug", "--block")
+	env.PrintStep("Starting %v service", gravityResumeServiceName)
+	if err := launchOneshotService(gravityResumeServiceName, args); err != nil {
+		return trace.Wrap(err)
+	}
+	env.PrintStep(`Service %[1]v has been launched.
+
+To monitor the operation progress:
+
+  sudo gravity plan --operation-id=%[2]v --tail
+
+To monitor the service logs:
+
+  sudo journalctl -u %[1]v -f
+`, gravityResumeServiceName, operation.ID)
+	return nil
+}
+
+// launchOneshotService launches the specified command as a one-shot systemd
+// service with the specified name.
+func launchOneshotService(name string, args []string) error {
```

Comment:
I wonder how that differs from https://github.com/gravitational/gravity/blob/version/5.5.x/tool/gravity/cli/system.go#L313

Reply:
Yeah, we have a couple of similar methods which I originally tried to see if they will fit this use-case, but they're different in a couple of ways. This method, for example, makes sure there's no another service running to prevent resume from being launched multiple times simultaneously.